.github/workflows/master-e2e.yaml

```diff
@@ -290,12 +300,91 @@ jobs:
           RSA_PRIVATE_KEY_QA: ${{ secrets.rsa_private_key_qa }}
           RSA_PUBLIC_KEY_QA: ${{ secrets.rsa_public_key_qa }}
           GREPTAGS: ${{ inputs.grep_test_by_tag }}
+          UPGRADE: 'false'
           # Add cypress/e2e/unit_tests/user.spec.ts again when implementing RBAC tests.
           SPEC: |
             cypress/e2e/unit_tests/first_login_rancher.spec.ts
             cypress/e2e/unit_tests/p0_fleet.spec.ts
             cypress/e2e/unit_tests/p1_fleet.spec.ts
             cypress/e2e/unit_tests/rbac_fleet.spec.ts
+            cypress/e2e/unit_tests/upgrade_fleet.spec.ts
+        run: |
+          if ${{ inputs.qase_run_id == 'none' }}; then
+            # Unset default QASE_* variables when reporting is disabled
+            unset QASE_REPORT
+            unset QASE_API_TOKEN
+            # QASE_RUN_ID is empty string already
+          fi
+          cd tests && make start-cypress-tests
+      - name: Upgrade Rancher Manager
+        id: rancher_upgrade
+        if: ${{ inputs.rancher_upgrade != '' }}
+        env:
+          PUBLIC_DNS: ${{ needs.create-runner.outputs.public_dns }}
+          PUBLIC_DOMAIN: bc.googleusercontent.com
+          RANCHER_UPGRADE: ${{ inputs.rancher_upgrade }}
+        run: |
+          cd tests
+
+          if make e2e-upgrade-rancher-manager; then
+            # Extract Rancher Manager version
+            RANCHER_VERSION=$(kubectl get pod \
+                                --namespace cattle-system \
+                                -l app=rancher \
+                                -o jsonpath={.items[*].status.containerStatuses[*].image} 2> /dev/null || true)
+
+            # Export values
+            echo "rancher_image_version=${RANCHER_VERSION}" >> ${GITHUB_OUTPUT}
+
+          else
+            # Needed to be sure that Github Action will see the failure
+            false
+          fi
+      - name: Extract component versions/informations after Upgrade
+        id: upgraded_component
+        run: |
+          # Extract Rancher Manager version
+          RM_UPGRADED_VERSION=$(kubectl get pod \
+            --namespace cattle-system \
+            -l app=rancher \
+            -o jsonpath={.items[*].status.containerStatuses[*].image} 2> /dev/null || true)
+          # Extract Fleet App version and images from local cluster
+          FLEET_APP_UPGRADED_VERSION="$(helm list -n cattle-fleet-system -o json 2> /dev/null \
+            | jq -r '.[] | .chart' \
+            | sort -V \
+            | uniq \
+            | tail -n 1)"
```

Comment:
```suggestion
          FLEET_APP_UPGRADED_VERSION="$(helm list -n cattle-fleet-system -o json 2> /dev/null | | jq -r '.[0] | .chart"
```

Reply:
I tried that one, but it was printing all version including the previous versions too...
So, I came up with `tail -n 1` part. But Happy to do in follow-up PR if you're okay?

Reply:
I updated in the same patchset.